addon/adapters/adapter.js

```diff
@@ -156,6 +173,46 @@ export default Adapter.extend(ImportExportMixin, {
     return null;
   },
 
+  _queryFilter(record, query = {}) {
+    return keys(query).every((key) => {
+      const value = query[key],
+        matches = key.match(/(.*)Id$/),
```

Comment:
You are right. I'm not sure if I got your suggestion right. Is this what you have in mind:
```
// get posts of a user
query('post', filter: { user: { id: '123' } });

// get users with post.id = 123
query('user', filter: { posts: { id: '123' } });

// get users with cats
query('user', filter: { pets: { type: 'cat' } });
```

On second thought it seems to me that querying on hasMany relations only makes sense for polymorphic relations. I could not think of a case where `query('user', filter: { posts: { id: '123' } })` can help you.

What do you think?